src/pytest_inmanta_lsm/plugin.py

```diff
@@ -282,25 +277,31 @@ def remote_orchestrator_shared(
         ):
             LOGGER.warning("SSL currently doesn't work with the default docker-compose file.")
 
-    token: Optional[str]
-    try:
-        token = inm_lsm_token.resolve(request.config)
-    except ParameterNotSetException:
-        token = None
+    def get_optional_option(option: StringTestParameter) -> Optional[str]:
+        try:
+            return option.resolve(request.config)
+        except ParameterNotSetException:
+            return None
+
+    token = get_optional_option(inm_lsm_token)
+    environment_name = get_optional_option(inm_lsm_env_name)
+    project_name = get_optional_option(inm_lsm_project_name)
 
     remote_orchestrator = RemoteOrchestrator(
+        module.Project(project_shared._test_project_dir),
```

Comment:
Conclusion: I would definitely use `Project.get()` because `Project` has a bunch of parameters that are set by `pytest-inmanta` and may be changed in the future. The venv being key among them.

As to when to call it, the most stable approach would be to do so dynamically because pytest-inmanta creates a new one for each compile. If you think this would be undesirable I can give the potential consequences some more thought but I think we'd need to step very carefully.

Reply:
I updated the logic this way.  Is there a specific error I could catch and a good explanation I could give when `Project.get()` doesn't "work"?  Given that it is not something that the user provides, it looks like something it does not have control over (at least through the `RemoteOrchestrator` class) so giving some pointers on what is going on and how it could be fixed could be nice.

Reply:
Alas it doesn't raise an exception if no project has been configured, instead it creates one and sets it. You could check for `project.loaded`. I think all calls into the remote orchestrator expect a project to be loaded already, right?

Reply:
It is expected indeed, I will check for project.loaded and log a warning if it is not then